sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishingEvents.md

```diff
@@ -48,13 +56,65 @@ finally
 }
 ```
 
-The `EventDataBatch` is scoped to a single publish operation.  Once that operation is complete, a new batch should be created for any additional events to be published.  The batch is responsible for unmanaged resources; it is recommended that you `Dispose` the batch after it has been published.
+The `EventDataBatch` is scoped to a single publish operation.  Once that operation is complete, a new batch should be created for any additional events to be published.  Because the batch is responsible for unmanaged resources, it is recommended that you `Dispose` the batch after it has been published.
+
+## Publishing and partitions
+
+Every event that is published is sent to one of the [partitions](https://docs.microsoft.com/azure/architecture/reference-architectures/event-hubs/partitioning-in-event-hubs-and-kafka) of the Event Hub. The application may request publishing to a specific partition, grouped using a partition key,  or allow the partition to be chosen automatically.  
+
+When using the `EventHubProducerClient`, each batch must choose the partition assignment strategy at the time it is created, and that strategy is applied to all events in the batch.   The `EventHubBufferedProducerClient` allows the partition assignment strategy to be chosen for each individual event that is enqueued, and the producer will ensure that batches are constructed with the proper strategy.
 
 ## Publishing events with automatic partition assignment
 
 Allowing automatic assignment to partitions is recommended when publishing needs to be highly available and shouldn't fail if a single partition is experiencing trouble.  Automatic assignment also helps to ensure that event data is evenly distributed among all available partitions, which helps to ensure throughput when publishing and reading data.
 
-When the batch is published, the `EventHubProducerClient` will receive an acknowledgment from the Event Hubs service; so long as no exception is thrown by this call, your application can consider publishing successful.  The service assumes responsibility for delivery of the batch.  All of your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
+### Event Hub Buffered Producer Client
+
+When using the `EventHubBufferedProducerClient`, events enqueued with no options specified will be automatically routed.  Because the producer manages publishing, there is no explicit call.  When the producer is closed, it will ensure that any remaining enqueued events have been published.  All of your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
```

Comment:
```suggestion
When using the `EventHubBufferedProducerClient`, events enqueued with no options specified will be automatically routed.  Because the producer manages publishing, there is no explicit call.  When the producer is closed, it will ensure that any remaining enqueued events have been published.  All of your event data will be published to one of the Event Hub partitions, though there may be a slight delay until it is available to be read.
```